### PR TITLE
[onert-micro] Introduce OMTrainingStorage and OMTrainingHandler

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr introduces OMTrainingStorage and OMTrainingHandler entities.

for issue https://github.com/Samsung/ONE/issues/12873
from draft: https://github.com/Samsung/ONE/pull/13107

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+    }
+    default:
+    {
+      assert(false && "Unsupported type");
+      return UnsupportedType;
+    }
+  }
+
+  return status;
+}
```
Comment (ljwoo94) [inline onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp:145]: If we make a map for 
{ config.training_context.optimizer : std::unique_ptr<opimizer>... }
I think we can remove all these switch cases and code dependencies like `.getSGD`, `getAdam`... for later extensibility.

Does it have a risk? or costs?

Comment (BalyshevArtem) [reply onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp:145]: Yes, I think it is good idea, thank you. I will try it now and push this changes :)

Comment (BalyshevArtem) [reply onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp:145]: @ljwoo94 
Ah to save it in `std::unordered_map<optimizer_type , std::unique_ptr<Optimizer>>` I need common `Optimizer` type and virtual mechanism to call right functions. But in onert-micro we disable virtual mechanism in order to save binary. 
So, I think such switch case is the most readable way to use different optimizers :(
 Or do you have some ideas?

**Review by Torrero — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Review by ljwoo94 — APPROVED** [review]

File: onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp (hunk 1 above)
Comment (ljwoo94) [reply onert-micro/onert-micro/src/core/train/OMTrainingHandler.cpp:145]: > Or do you have some ideas?
I guess not :'(

Approved PR. LGTM :)

**Pull request merged**